bears/general/LineBreakBear.py

```diff
@@ -0,0 +1,135 @@
+        :param tab_width:
+            No. of spaces to add before the newly created line.
+        :param max_line_length:
+            Maximum number of characters for a line, the newline character
+            being excluded.
+        :param dependency_results:
+            Results given by the AnnotationBear.
+        :param coalang_dir:
+            Full path of external directory containing the coalang
+            file for language.
+        """
+        lang_settings_dict = LanguageDefinition(
+            language, coalang_dir=coalang_dir)
+
+        encapsulators = dict(lang_settings_dict["encapsulators"])
+
+        # Get all lines which are over max_line_length
+        line_length_bear = LineLengthBear(self.section, Queue())
+        line_length_results = list(line_length_bear.execute(filename, file))
+
+        affected_lines = []
+        for line_length_result in line_length_results:
+            for _range in line_length_result.affected_code:
+                affected_lines.append(_range.start.line)
+
+        annotation_dict = dependency_results[AnnotationBear.name][0].contents
+
+        # Get all positions of encapsulators
+        encaps_pos = []
+        for encapsulator in encapsulators:
+            encaps_pos += get_specified_block_range(
+                file, filename,
+                encapsulator, encapsulators[encapsulator],
+                annotation_dict)
+        encaps_pos = tuple(sorted(encaps_pos, key=lambda x: (x.start.line,
+                                                             x.start.column)))
+
+        insert = indentation_width*" " if use_spaces else "\t"
+
+        suggested_positions = self._find_breakable_encapsulators(
+            file,
+            filename,
+            affected_lines,
+            max_line_length,
+            encaps_pos)
+        new_file = self._break_after_position(file,
+                                              insert,
+                                              suggested_positions)
+        if new_file != list(file):
+            wholediff = Diff.from_string_arrays(file, new_file)
+            for diff in wholediff.split_diff():
+                yield Result(
+                    self,
+                    'Following is a suggestion to break this long line',
+                    severity=RESULT_SEVERITY.INFO,
+                    affected_code=(diff.range(filename),),
+                    diffs={filename: diff})
+
+    def _break_after_position(self,
+                              file,
+                              insert,
+                              suggested_positions):
+        new_file = list(file)
+        for position in suggested_positions:
+            new_line = (insert +
+                        file[position.line - 1][position.column:])
+            new_file.insert(position.line, new_line)
+            new_file[position.line - 1] = new_file[position.line - 1][
+                :position.column] + '\n'
+        return new_file
```

Comment:
this looks good for basic cases, but there a probably a lot of corner cases you need to take care of, hm? hanging indents and stuff? Looks good for now though.